mapie/subsample.py

```diff
@@ -101,3 +101,138 @@ def get_n_splits(self, *args: Any, **kargs: Any) -> int:
+    It can replace KFold, LeaveOneOut or SubSample as cv argument in the MAPIE
+    class.
+
+    Parameters
+    ----------
+    n_resamplings : int
+        Number of resamplings. By default ``30``.
+    length: int
+        Length of the blocks. By default ``None``,
+        the length of the training set divided by ``n_blocks``.
+    overlapping: bool
+                Whether the blocks can overlapp or not. By default ``False``.
+    n_blocsk: int
+        Number of blocks in each resampling. By default ``None``,
+        the size of the training set divided by ``length``.
+    random_state: Optional
+        int or RandomState instance.
+
+    Raises
+    ------
+    ValueError
+        If both ``length`` and ``n_blocks`` are ``None``.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from mapie.subsample import BlockBootstrap
+    >>> cv = BlockBootstrap(n_resamplings=2, length = 3, random_state=0)
+    >>> X = np.array([1,2,3,4,5,6,7,8,9,10])
+    >>> for train_index, test_index in cv.split(X):
+    ...    print(f"train index is {train_index}, test index is {test_index}")
+    train index is [1 2 3 4 5 6 1 2 3 4 5 6], test index is [8 9 7]
+    train index is [4 5 6 7 8 9 1 2 3 7 8 9], test index is []
+    """
+
+    def __init__(
+        self,
+        n_resamplings: int = 30,
+        length: Optional[int] = None,
+        n_blocks: Optional[int] = None,
+        overlapping: bool = False,
+        random_state: Optional[Union[int, RandomState]] = None,
+    ) -> None:
+        self.n_resamplings = n_resamplings
+        self.length = length
+        self.n_blocks = n_blocks
+        self.overlapping = overlapping
+        self.random_state = random_state
+
+    def split(
+        self, X: NDArray
+    ) -> Generator[Tuple[NDArray, NDArray], None, None]:
+        """
+        Generate indices to split data into training and test sets.
+
+        Parameters
+        ----------
+        X : NDArray of shape (n_samples, n_features)
+            Training data.
+
+        Yields
+        ------
+        train : NDArray of shape (n_indices_training,)
+            The training set indices for that split.
+        test : NDArray of shape (n_indices_test,)
+            The testing set indices for that split.
+        Raises
+        ------
+        ValueError
+            If ``length`` is not positive or greater than the train set size.
+        """
+        if self.n_blocks is not None:
+            length = (
+                self.length
+                if self.length is not None
+                else len(X) // self.n_blocks
+            )
+            n_blocks = self.n_blocks
+        elif self.length is not None:
+            length = self.length
+            n_blocks = (len(X) // self.length) + 1
+        else:
+            raise ValueError(
+                "At least one argument between ``length`` or "
+                "``n_blocks`` has to be not None"
+            )
+        indices = np.arange(len(X))
```

Comment:
Add `n = len(X)` that can be used also for `len(indices)`

Reply:
Done